### DISCUSSION TITLE
Please allow reusable workflows in private repositories

### DISCUSSION THREAD

**Discussion by peterfigure** [community/community#9445] [Actions] [answered]
According to the [documentation](https://docs.github.com/en/actions/learn-github-actions/reusing-workflows#access-to-reusable-workflows):

A reusable workflow can be used by another workflow if either of the following is true:
* Both workflows are in the same repository.
* The called workflow is stored in a public repository.

Unless I'm missing something, this is pretty much a nonstarter for any reasonable organization. 

Eg, we have over 800 repositories, so the first criteria isn't viable (reusing workflows to us means all of the repositories reusing workflows defined centrally in a single or handful of repositories, DRY etc). 
The second criteria isn't viable either, it's not information we want public, I don't think that's unreasonable. 

Is this on the roadmap or are there any known workarounds? Briefly looking around it doesn't look like Git submodules works either 😢

**Comment by chrispat** [comment, marked as answer]
Thank you for your feedback.

This feature is [on our roadmap](https://github.com/github/roadmap/issues/74) and should be shipping soon.

**Reply by peterfigure** [reply]
Great to hear, thank you!

**Reply by andreitsibets** [reply]
But the feature on roadmap is for `internal` repositories, while request is for `private` repositories.

**Reply by kriswuollett** [reply]
I was curious and tried following the instructions at https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-a-docker-hub-action but used `${{ secrets.GITHUB_TOKEN }}` instead of a personal access token as suggested. It doesn't work because git says repo not found when trying to do the clone, but clearly it is there. That leads me to believe that this can be solved fairly quickly if GitHub would add access to all repos within an organization to the principal represented by the automatic `GITHUB_TOKEN` secret. Referencing the private github repo on the `uses:` line is the extra nice to have.

**Comment by marill** [comment]
Will this be available for GitHub Team accounts too, or only Enterprise?

**Comment by petelah** [comment]
Any update on this?

**Comment by MaxStroh** [comment]
We would be interested in this, too. Especially, because the referenced issue is not about what was targeted / requested here. https://github.com/github/roadmap/issues/74 only states about shared actions, while even sharing whole reusable workflows would be very great and useful.

**Comment by pragmaticivan** [comment]
This seems to be ready for enterprise: https://docs.github.com/en/enterprise-cloud@latest/actions/creating-actions/sharing-actions-and-workflows-with-your-enterprise

**Reply by aavileli** [reply]
Make it available for teams .

**Comment by prazian** [comment]
Why this is only available for Enterprise, but not Team? :/

Many companies use the Team subscription because they don't need the extra things in the Enterprise subscription, and in my opinion, not having this basic feature in the Team subscription demotes DRY in the software world.

**Reply by ideas-man** [reply]
Exactly. My org doesn't have a big team, but enough repos, so we really need proper automation and DRY here. We also can't really afford making anything public, so this really hurts us. I would expect this basic feature available for Teams, too.

**Comment by freemanjohn** [comment]
Individual attempting to move from an internal hosted gitlab container instance, to github, however it looks like the feature sets even when comparing against gitlab-ce are not there... Sad as was hoping could utilize github and move away the company repos/configuration, but this an others issues are obvious non starters.

**Comment by fishpen0** [comment]
How does the community vote to unmark an answer given the current answer references a closed issue and this was never actually fixed for private repos? @chrispat 

This looks to be on the roadmap but keeps getting bumped: https://github.com/github/roadmap/issues/51
